src/Bicep.IO/FileSystem/FileSystemFileHandle.cs

```diff
@@ -43,6 +43,23 @@ public Stream OpenWrite()
             return this.FileSystem.FileStream.New(this.FilePath, FileMode.Create, FileAccess.Write, FileShare.None);
         }
 
+        public string ReadAllText() => this.FileSystem.File.ReadAllText(this.FilePath);
+
+        public Task<string> ReadAllTextAsync() => this.FileSystem.File.ReadAllTextAsync(this.FilePath);
+
+        public void WriteAllText(string text)
+        {
+            this.GetParent().EnsureExists();
+            this.FileSystem.File.WriteAllText(this.FilePath, text);
+        }
+
+        public async Task WriteAllTextAsync(string text)
+        {
+            this.GetParent().EnsureExists();
+
+            await this.FileSystem.File.WriteAllTextAsync(this.FilePath, text);
+        }
+
```

Comment:
Using these methods instead of writing our own implementation using `Stream` because .NET’s built-in versions are highly optimized (e.g., `File.WriteAllText` uses `SafeFileHandle` and `RandomAccess`). Replicating the same level of performance with Stream isn’t feasible.